src/Workspaces/Core/Portable/Shared/Extensions/SyntaxGeneratorExtensions_CreateGetHashCodeMethod.cs

```diff
@@ -37,6 +36,40 @@ internal static partial class SyntaxGeneratorExtensions
             return result.ToImmutableAndFree();
         }
 
+        public static ImmutableArray<SyntaxNode> CreateGetHashCodeStatementsUsingSystemHashCode(
+            this SyntaxGenerator factory, INamedTypeSymbol hashCodeType, ImmutableArray<SyntaxNode> memberReferences)
+        {
+            if (memberReferences.Length <= 8)
+            {
+                var statement = factory.ReturnStatement(
+                    factory.InvocationExpression(
+                        factory.MemberAccessExpression(factory.TypeExpression(hashCodeType), "Combine"),
+                        memberReferences));
+                return ImmutableArray.Create(statement);
+            }
+
+            const string hashName = "hash";
```

Comment:
I didn't see any tests for this case, did I miss them?

Reply:
will add.